ciw/tests/test_state_tracker.py

```diff
@@ -1367,4 +1367,67 @@ def test_error_checking_for_state_probabilities(self):
         self.assertRaises(ValueError, Q.statetracker.state_probabilities, (-1, 5))
         self.assertRaises(ValueError, Q.statetracker.state_probabilities, (4, 2))
         self.assertRaises(ValueError, Q.statetracker.state_probabilities, (-1, -4))
-        self.assertRaises(ValueError, Q.statetracker.state_probabilities, (3, 3))
\ No newline at end of file
+        self.assertRaises(ValueError, Q.statetracker.state_probabilities, (3, 3))
+
+
+    def test_state_tracker_with_reneging(self):
+        N = ciw.create_network(
+            arrival_distributions=[ciw.dists.Deterministic(7)],
+            service_distributions=[ciw.dists.Deterministic(11)],
+            number_of_servers=[1],
+            reneging_time_distributions=[ciw.dists.Deterministic(3)]
+        )
+        Q = ciw.Simulation(N, tracker=ciw.trackers.SystemPopulation())
+        self.assertTrue(Q.nodes[1].reneging)
+        #### We would expect:
+        # t=0  begin          ---> 0
+        # t=7  arrival cust 1 ---> 1
+        # t=14 arrival cust 2 ---> 2
+        # t=17 renege  cust 2 ---> 1
+        # t=18 leave   cust 1 ---> 0
+        # t=21 arrival cust 3 ---> 1
+        # t=28 arrival cust 4 ---> 2
+        # t=31 renege  cust 4 ---> 1
+        # t=32 leave   cust 3 ---> 0
```

Comment:
I'd suggest moving this to the docstring of `def test_state_tracker_with_reneging` and add a bit more context for clarity.